### ISSUE TITLE
Unable to configure GCP's zonal NEGs by Halyard

### ISSUE THREAD

**Issue by KeisukeYamashita** [spinnaker/spinnaker#5934]
### Issue Summary:

Because the Halyard won't allow us to configure custom service annotation, we can't enable [Network Endpoint Group(NEG)](https://cloud.google.com/load-balancing/docs/negs/zonal-neg-concepts) for [GKE](https://cloud.google.com/kubernetes-engine).

### Cloud Provider(s):

* All services that have `kind: Service` Kubernetes resource

### Environment:

* Google Cloud Platform(GCP)'s Google Kubernetes Engine(GKE)

### Feature Area:

* Halyard

### Description:

According to Spinnaker's [Custom Conifguration](https://spinnaker.io/reference/halyard/custom/), only [podAnnotations, podLabels and serviceLabels](https://spinnaker.io/reference/halyard/custom/#podannotations-podlabels-and-servicelabels) are configurable and there are no serviceAnnotations.

To enable NEG on GKE, I need to create a Kubernetes Service like below(quote from GCP's [guide](https://cloud.google.com/kubernetes-engine/docs/how-to/standalone-neg)):

```yaml
apiVersion: v1
kind: Service
metadata:
  name: neg-demo-svc
  annotations:
    cloud.google.com/neg: '{"exposed_ports": {"80":{}}}'   <---- I need to add this annotation to enable NEG
spec:
  type: ClusterIP
  selector:
    run: neg-demo-app # Selects Pods labelled run: neg-demo-app
  ports:
  - port: 80
    protocol: TCP
    targetPort: 9376
```

NEG enables container-native load balancing which there are many great benefits.(See detail in the GCP docs).
Therefore, I want to enable this though Halyard.

### Steps to Reproduce:

* Try to configure custom configuration by `kubernetes.serviceAnnotations`. This will not work because there are no implementations.

### Additional Details:

* This PR will resolve the issue. https://github.com/spinnaker/halyard/pull/1763

**Comment by ezimanyi** [comment]
Addressed in spinnaker/halyard#1763.

**Comment by KeisukeYamashita** [comment]
Note: The document updating PR is here → https://github.com/spinnaker/spinnaker.github.io/pull/1988
